app/models/host/managed.rb

```diff
@@ -655,6 +656,17 @@ def clone
     host
   end
 
+  def check_interfaces
+    self.interfaces.detect(&:primary).nil? ? missing_primary = true : missing_primary = false
+    self.interfaces.detect(&:provision).nil? ? missing_provision = true : missing_provision = false
+
+    if missing_provision
+      errors.add(:base, _("An interface marked as provision is missing"))
+    elsif missing_primary
+      errors.add(:base, _("An interface marked as primary is missing"))
+    end
+  end
```

Comment:
First, use `else` instead of `elseif` because the Rails error stack can hold multiple errors. Why not to show them all to the user when we know it's missing?

Second and this is a bit of nitpicking - can you just get rid of helper variables which makes the method quite long, you can do just `if self.interfaces.detect(&:primary).nil?` just fine, this will be shorter yet still readable.

Reply:
@lzap  I cannot use `else` because if primary is not missing that doesn't automatically means provision interface is missing. I can absolutely rid of checking provision interface if you want( provision is automatically set in next steps)

Reply:
@lzap I cannot use `else` because if primary is not missing that doesn't automatically means provision interface is missing. I can absolutely rid of checking provision interface if you want( provision is automatically set in next steps)

Reply:
@lzap I cannot use else because if primary is not missing that doesn't automatically means provision interface is missing. I can absolutely rid of checking provision interface if you want( provision is automatically set in next steps)

Reply:
Sorry I meant not to use `else` at all, two ifs. Let me show:

```ruby   
     errors.add(:base, _("An interface marked as provision is missing")) if if self.interfaces.detect(&:provision).nil?
     errors.add(:base, _("An interface marked as primary is missing")) if self.interfaces.detect(&:primary).nil?
```